fusestore/store/src/executor/meta_handlers.rs

```diff
@@ -91,51 +109,78 @@ impl RequestHandler<DropDatabaseAction> for ActionHandler {
+
+        match rst {
+            AppliedState::DataBase { .. } => Ok(DropDatabaseActionResult {}),
+            _ => Err(ErrorCode::MetaNodeInternalError("not a Database result")),
+        }
     }
 }
-// table
 
+// table
 #[async_trait::async_trait]
 impl RequestHandler<CreateTableAction> for ActionHandler {
     async fn handle(
         &self,
         act: CreateTableAction,
     ) -> common_exception::Result<CreateTableActionResult> {
         let plan = act.plan;
-        let db_name = plan.db;
-        let table_name = plan.table;
+        let db_name = &plan.db;
+        let table_name = &plan.table;
 
-        info!("create table: {:}: {:?}", db_name, table_name);
+        info!("create table: {:}: {:?}", &db_name, &table_name);
 
-        let mut meta = self.meta.lock();
+        // if database not exists, return early.
+        self.meta_node.get_database(db_name).await.ok_or_else(|| {
+            ErrorCode::UnknownDatabase(format!("create table: database not found {:}", db_name))
+        })?;
 
         let options = common_arrow::arrow::ipc::writer::IpcWriteOptions::default();
         let flight_data: FlightData =
             arrow_flight::SchemaAsIpc::new(&plan.schema.to_arrow(), &options).into();
 
         let table = Table {
-            // the storage engine fills the id.
-            table_id: -1,
-            ver: -1,
+            table_id: 0,
             schema: flight_data.data_header,
-            options: plan.options,
-
-            // TODO
-            placement_policy: vec![],
+            parts: Default::default(), // TODO(ariesdevil): what's this?
         };
 
-        let cmd = CmdCreateTable {
-            db_name,
-            table_name,
-            table: Some(table),
+        let cr = LogEntry {
+            txid: None,
+            cmd: CreateTable {
+                db_name: db_name.clone(),
+                table_name: table_name.clone(),
+                if_not_exists: plan.if_not_exists,
+                table,
+            },
         };
 
-        let table_id = meta.create_table(cmd, plan.if_not_exists)?;
+        let rst = self.meta_node.write(cr).await.map_err(|e| {
+            ErrorCode::MetaNodeInternalError(format!("not a Table result: {}", e.to_string()))
+        })?;
 
-        Ok(CreateTableActionResult { table_id })
+        match rst {
+            AppliedState::Table { result, .. } if result.is_some() => Ok(CreateTableActionResult {
+                table_id: result.unwrap().table_id,
+            }),
+            AppliedState::Table { prev, .. } if prev.is_some() => Ok(CreateTableActionResult {
+                table_id: prev.unwrap().table_id,
+            }),
+            AppliedState::Table { prev, result } if (prev.is_none() && result.is_none()) => Err(
+                ErrorCode::TableAlreadyExists(format!("table exists: {}", table_name)),
+            ),
+            _ => Err(ErrorCode::MetaNodeInternalError("not a Table result")),
```

Comment:
The pattern `_` matches negative conditions: If it is a `AppliedState::Table { result, .. }` but result is None, it should not return an error `not a Table result`.